part/set.go

```diff
@@ -121,11 +116,41 @@ func (s Set[T]) Len() int {
 	return s.tree.size
 }
 
+// Equal returns true if the two sets contain the equal keys.
+func (s Set[T]) Equal(other Set[T]) bool {
+	switch {
+	case s.tree == nil:
+		return other.tree == nil
+	case other.tree == nil:
+		return s.tree == nil
```

Comment:
or just `false`.

Reply:
No it needs to return true if both sets are empty, but I think I'll simplify this a bit.